### PR TITLE
#176 User Management: Editing User

### PR DESCRIPTION
**Pull request by a2batic**
Bug-id: #176
Signed-off-by: a2batic <murarkakanika@gmail.com>

### CODE REVIEW

**Comment by a2batic** [comment]
@anivargi @shtripat @nthomas-redhat @brainfunked @gnehapk @julienlim @cloudbehl , please review

File: specs/Spec_for_editing_user.adoc [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+User will land onto https://redhat.invisionapp.com/share/KNB25OEQT#/screens/226063805[Edit User] view page
+
+## Proposed workflow
+
+### UI
+
+Edit user view consist of a form:-
+
+* ```User ID``` - Not Editable
+* ```Type : Internal``` - Not Editable
```
Comment (shtripat) [inline specs/Spec_for_editing_user.adoc]: What is type internal? Is it like `admin` is internal user and cannot be edited?

File: specs/Spec_for_editing_user.adoc [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+* Password
+** Input type: password
+** Validation: Should be minimum of 8 characters
+** Should contain a Show/Hide button for password visibility. The default visibility will be *hidden*.
+* Select a Role
+** Roles: Admin, Normal, Limited
+** The default will be the current assigned role of User
+* ```Save``` button
+** Enabled when User makes change in this form.
+** This button commits changes and returns to User List view page.
```
Comment (shtripat) [inline specs/Spec_for_editing_user.adoc]: We should show a messages as well that user edit was successful as a balloon I feel. Do we have that?

File: specs/Spec_for_editing_user.adoc (hunk 1 above)
Comment (a2batic) [reply specs/Spec_for_editing_user.adoc]: It is related to LDAP, so has to be removed, I will update it

File: specs/Spec_for_editing_user.adoc (hunk 2 above)
Comment (a2batic) [reply specs/Spec_for_editing_user.adoc]: @gnehapk, @brainfunked

Comment (a2batic) [reply specs/Spec_for_editing_user.adoc]: @julienlim , @mcarrano, Can you please suggest if there should be a pop up or notification for this?

File: specs/Spec_for_editing_user.adoc [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+API support present for -
+
+* email
+* username
+* name
+* role
+
+Required from API - 
+
+* first_name
```
Comment (cybernth) [inline specs/Spec_for_editing_user.adoc]: Per https://github.com/Tendrl/specifications/pull/235/commits/819f0a8b7323a6c72ca37d7864cfea6973a8583a#diff-047658fc19cc3b6a6cba0d6749752529R65,  you need to concatenate first name and last name and send across to API?

File: specs/Spec_for_editing_user.adoc [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+* last_name
+* status
```
Comment (cybernth) [inline specs/Spec_for_editing_user.adoc]: are we going to support the status?

**Review by cybernth — CHANGES_REQUESTED** [review]

File: specs/Spec_for_editing_user.adoc (hunk 3 above)
Comment (a2batic) [reply specs/Spec_for_editing_user.adoc]: Yes, I have to update that

File: specs/Spec_for_editing_user.adoc (hunk 4 above)
Comment (a2batic) [reply specs/Spec_for_editing_user.adoc]: @anivargi ^^

File: specs/Spec_for_editing_user.adoc (hunk 2 above)
Comment (julienlim) [reply specs/Spec_for_editing_user.adoc]: @a2batic No pop-up or notification needed.

**Pull request merged**